Library/Homebrew/tap.rb

```diff
@@ -96,6 +96,31 @@ def command_files
     Pathname.glob("#{path}/cmd/brew-*").select(&:executable?)
   end
 
+  def pinned_symlink_path
+    (HOMEBREW_LIBRARY/"PinnedTaps/#{@name}")
+  end
+
+  def pinned?
+    @pinned ||= pinned_symlink_path.directory?
+  end
+
+  def pin
+    begin
+      pinned_symlink_path.make_relative_symlink(@path)
+    rescue SystemCallError
+      oppo "Something went wrong." # TODO
```

Comment:
Use `raise ...`(you can create new exception in `exceptions.rb`) instead of `opoo`.